Code/Model.py

```diff
@@ -0,0 +1,174 @@
+"""
+    @file:              DataGen.py
+    @Author:            Nicolas Raymond
+    @Creation Date:     30/09/2019
+    @Last modification: 30/09/2019
+    @Description:       This program generate models of different types to use for our classification problems.
+
+"""
+
+import numpy as np
+import sklearn as sk
+
+class Model:
+
+    def __init__(self, HP_Dict):
+
+        """
+        Class that generate a model to solve a classification problem
+
+        :param HP_Dict: Dictionary containing all hyperparameters of the model
+
+        """
+        HP_space = HP_Dict
```

Comment:
self.HP_space

?

Reply:
It defines the space of every of our hyperparameters. More precisely, every hyperparam specified in the dictionary could be associate to a single value, a list of possible values or even a statical distribution which represents every possible values of the hyperparameter.
At the start, only a single value is associate to a hyperparameter "key" in our dictionnary. But the search space will be defined in the HPtuner class. A concept of HPtuner is in development in another branch.